client/turn_done_button.cpp

```diff
@@ -130,12 +130,38 @@ void turn_done_button::paintEvent(QPaintEvent *event)
 
 namespace {
 
+/**
+ * Formats a duration without switching to "until hh::mm" when more than one
+ * hour in the future.
+ */
+QString format_simple_duration(int seconds)
+{
+  if (seconds < 0) {
+    seconds = 0;
+  }
+
+  if (seconds < 60) {
+    return QString(Q_("?seconds:%1s")).arg(seconds, 2);
+  } else if (seconds < 5 * 60) { // < 5 minutes
+    return QString(Q_("?mins/secs:%1min %2s"))
+        .arg(seconds / 60, 2)
+        .arg(seconds % 60, 2);
+  } else if (seconds < 3600) { // < one hour
+    // TRANS: Used in "Time left: 10 minutes". Always at least 5 minutes
+    return QString(_("%1 minutes")).arg(seconds / 60);
+  } else {
+    // Hours and minutes
+    const auto minutes = seconds / 60;
+    return QStringLiteral("%1:%2").arg(minutes / 60).arg(minutes % 60);
```

Comment:
Let's add h and m - `QStringLiteral("%1h:%2m)` to follow similar formatting of the others above.

Reply:
`%1h %2min` then?

Reply:
yea that is perfect